src/commcare_cloud/commands/terraform/templates/commcarehq.tf.j2

```diff
@@ -317,3 +317,49 @@ module "efs_mount__{{ efs_file_system_mount.efs_name }}__{{ az }}" {
 {% endfor %}
 
 {% endif %}
+
+{% if existing_zone_record_update %}
+{% for r53_record_list in existing_zone_record_update.records_list %}
+module "r53-record__{{ r53_record_list.route_names }}" {
+  source = "./modules/r53-record-create-update"
+  domain_name = {{ existing_zone_record_update.domain_name|tojson }}
+  route_names = {{ r53_record_list.route_names|tojson }}
+  record_type = {{ r53_record_list.record_type|tojson }}
+  records = [
+      {% if r53_record_list.route_names == 'rds_postgres' %}
+        {%- for rds_postgres_instances in r53_record_list.records %}{% if not loop.first %}, {% endif %}
+          "${module.postgresql__{{ rds_postgres_instances }}.postgresql_rds_this_db_instance_endpoint }"
+        {%- endfor %}
+      {% elif r53_record_list.route_names == 'commcarehq_synclogs' %}
+        {%- for commcarehq_synclogs_instance in r53_record_list.records %}{% if not loop.first %}, {% endif %}
+          "${module.server__{{ commcarehq_synclogs_instance }}.server_private_ip}"
+        {%- endfor %}
+      {% elif r53_record_list.route_names == 'proxy_form_processing' %}
+        {%- for proxy_form_processing_instance in r53_record_list.records %}{% if not loop.first %}, {% endif %}
+          "${module.server__{{ proxy_form_processing_instance }}.server_private_ip}"
+        {%- endfor %}
+      {% elif r53_record_list.route_names == 'couchdb2_proxy_host' %}
+        {%- for couchdb2_proxy_host_instance in r53_record_list.records %}{% if not loop.first %}, {% endif %}
+          "${module.server__{{ couchdb2_proxy_host_instance }}.server_private_ip}"
+        {%- endfor %}
+      {% elif r53_record_list.route_names == 'formplayer_pgbouncer' %}
+        {%- for pgbouncer_nlb_r53 in r53_record_list.records %}{% if not loop.first %}, {% endif %}
+          "${module.pgbouncer_nlb__{{ pgbouncer_nlb_r53 }}.aws_lb_dns_name }"
+        {%- endfor %}
+      {% elif r53_record_list.route_names == 'pgmain_nlb' %}
+        {%- for pgmain_nlb_r53 in r53_record_list.records %}{% if not loop.first %}, {% endif %}
+          "${module.pgbouncer_nlb__{{ pgmain_nlb_r53 }}.aws_lb_dns_name }"
+        {%- endfor %}
+      {% elif r53_record_list.route_names == 'pgsynclogs_nlb' %}
+        {%- for pgsynclogs_nlb_r53 in r53_record_list.records %}{% if not loop.first %}, {% endif %}
+          "${module.pgbouncer_nlb__{{ pgsynclogs_nlb_r53 }}.aws_lb_dns_name }"
+        {%- endfor %}
+      {% elif r53_record_list.route_names == 'pgucr_nlb' %}
+        {%- for pgucr_nlb_r53 in r53_record_list.records %}{% if not loop.first %}, {% endif %}
+          "${module.pgbouncer_nlb__{{ pgucr_nlb_r53 }}.aws_lb_dns_name }"
+        {%- endfor %}
```

Comment:
Continuing my comment about about the `target_type` idea.

I notice in this list (besides `rds_postgres` which I suggest we just delete), the logic follows one of two patters: either you look up `module.server__{{ target }}.server_private_ip`, or you look up `module.pgbouncer_nlb__{{ target }}.aws_lb_dns_name` resources. What I suggest is rather than spell out in this list every possible route_name we might come across, we simply look at `target_type` to decide how to look up the resource:

```
{% if existing_zone_record_update %}
{% for record in existing_zone_record_update.records_list %}
module "r53-record__{{ r53_record_list.route_names }}" {
  ...
  records = [
      {% if record.target_type == 'server' %}
        {%- for target in record.targets %}{% if not loop.first %}, {% endif %}
          "${module.server__{{ target }}.server_private_ip}"
        {%- endfor %}
      {% elif record.target_type == 'pgbouncer_nlb' %}
        {%- for target in record.targets %}{% if not loop.first %}, {% endif %}
          "${module.pgbouncer_nlb__{{ target }}.aws_lb_dns_name}"
        {%- endfor %}
      {% endif %}
  ]
}
{% endfor %}
{% endif %}
```

It's not only many fewer lines of code, but it also makes it so we can add other possibilities in terraform.yml without also updating this file.